src/Algebra/Structures.agda

```diff
@@ -862,10 +861,15 @@ record IsRing (+ * : Op₂ A) (-_ : Op₁ A) (0# 1# : A) : Set (a ⊔ ℓ) where
     )
 
   zeroˡ : LeftZero 0# *
-  zeroˡ = proj₁ zero
+  zeroˡ = Consequences.assoc+distribʳ+idʳ+invʳ⇒zeˡ setoid
+    +-cong *-cong +-assoc (proj₂ distrib) +-identityʳ -‿inverseʳ
```

Comment:
I think that there should be a definable `distribʳ` already in scope here, so you don't need to explicit projection?

Reply:
Yup moving the proof up the hierarchy has solved this problem.